app/models/manageiq/providers/vmware/infra_manager/ovf_service.rb

```diff
@@ -0,0 +1,150 @@
+class ManageIQ::Providers::Vmware::InfraManager::OvfService < ServiceGeneric
+  delegate :template, :manager, :to => :service_template, :allow_nil => true
+
+  # A chance for taking options from automate script to override options from a service dialog
+  def preprocess(action, new_options = {})
+    return unless action == ResourceAction::PROVISION
+
+    if new_options.present?
+      _log.info("Override with new options:")
+      $log.log_hashes(new_options)
+    end
+
+    save_action_options(action, new_options)
+  end
+
+  def execute(action)
+    return unless action == ResourceAction::PROVISION
+
+    library_item_deploy_queue
+  end
+
+  def library_item_deploy_queue
+    task_options = {:action => "Deploying VMware Content Library Item", :userid => "system"}
+    queue_options = {
+      :class_name  => self.class.name,
+      :instance_id => id,
+      :method_name => "library_item_deploy",
+      :args        => {},
+      :role        => "ems_operations",
+      #:queue_name  => manager.queue_name_for_ems_operations,
+      :zone        => manager.my_zone
+    }
+
+    task_id = MiqTask.generic_action_with_callback(task_options, queue_options)
+    task = MiqTask.wait_for_taskid(task_id)
+    raise task.message unless task.status_ok?
+  end
+
+  def library_item_deploy(_options)
+    _log.info("OVF template provisioning with template ID: [#{template.id}] name:[#{template.name}] was initiated.")
+    opts = provision_options
+    _log.info("VMware Content Library OVF Tempalte provisioning with options:")
+    $log.log_hashes(opts)
+
+    @deploy_response = template.deploy(opts).to_hash
+    _log.info("Content Library request response: #{@deploy_response}")
+    update(:options => options.merge(:deploy_response => @deploy_response))
+  rescue VSphereAutomation::ApiError => e
+    _log.error("Failed to deploy content library template(#{template.name}), error: #{e}")
+    raise MiqException::MiqOrchestrationProvisionError, "Content library OVF template deployment failed: #{e}"
+  end
+
+  def check_completed(action)
+    return [true, 'not supported'] unless action == ResourceAction::PROVISION
+    return [false, nil] if deploy_response.blank?
+
+    message = deploy_response.dig(:value, :succeeded) ? nil : deploy_response.dig(:value, :error).to_json || deploy_response.to_json
+    [true, message]
+  end
+
+  def refresh(_action)
+    EmsRefresh.queue_refresh(refresh_target) if deploy_response.dig(:value, :succeeded)
+  end
+
+  def check_refreshed(action)
+    return [true, nil] unless deploy_response.dig(:value, :succeeded)
+
+    dest = find_destination_in_vmdb
+    if dest
+      add_resource!(dest, :name => action)
+
+      if dest.kind_of?(ResourcePool)
+        dest.vms.each { |vm| add_resource!(vm, :name => action) }
+      end
+
+      [true, nil]
+    else
+      [false, nil]
+    end
+  end
+
+  def refresh_target
+    target_folder || target_host || target_resource_pool
```

Comment:
So we drop targeted refreshes anyway now that we're doing streaming refresh, https://github.com/ManageIQ/manageiq-providers-vmware/blob/master/app/models/manageiq/providers/vmware/infra_manager/refresh_worker/runner.rb#L22-L29

Reply:
I noticed this refresh was being skipped in the testing. Any change should be done here?

Reply:
Yeah you can drop the `queue_refresh` here since it is just going to be queued and dropped, now that we're in the vmware repo we can make the assumption that we are using streaming refresh.